src/components/tl/ucp/reduce_scatter/reduce_scatter_ring.c

```diff
@@ -223,13 +234,17 @@ ucc_tl_ucp_reduce_scatter_ring_start(ucc_coll_task_t *coll_task)
         return status;
     }
 
+    r_scratch  = task->reduce_scatter_ring.scratch;
     sendto     = ucc_ep_map_eval(task->reduce_scatter_ring.inv_map, sendto);
     recvfrom   = ucc_ep_map_eval(task->reduce_scatter_ring.inv_map, recvfrom);
-    r_scratch  = task->reduce_scatter_ring.scratch;
-    recv_block = (rank - 2 - step + size) % size;
-    send_block = (rank - 1 - step + size) % size;
     recv_block = ucc_ep_map_eval(task->reduce_scatter_ring.inv_map, recv_block);
     send_block = ucc_ep_map_eval(task->reduce_scatter_ring.inv_map, send_block);
+    if (team->cfg.use_reordering) {
```

Comment:
Here and elswhere we could simply define a map
```
ucc_ep_map_t map = team->cfg.use_reordering? task->subset.map : task->reduce_scatter_ring.inv_map)
```
and simply use this map instead of branching
What do you think ?

Reply:
no, in the case of bidirectional and reordering both maps are needed to be used one after the other.

Reply:
ok, got it